Objects/listobject.c

```diff
@@ -1022,21 +1022,28 @@ list_pop_impl(PyListObject *self, Py_ssize_t index)
         PyErr_SetString(PyExc_IndexError, "pop index out of range");
         return NULL;
     }
-    v = self->ob_item[index];
-    if (index == Py_SIZE(self) - 1) {
+
+    PyObject **items = self->ob_item;
+    v = items[index];
+    if(Py_SIZE(self)-1==0) {
```

Comment:
```suggestion
   Py_ssize_t pop_after_size = Py_SIZE(self) - 1;
    if(pop_after_size == 0) {
```

Reply:
> 

Good suggestion. I renamed to `size_after_pop` instead of the `pop_after_size`.